### PR TITLE
feat(dedicated.ip): configure game firewall - edit rule

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-483
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

- show line into edit rule mode
- add button to apply update rule
- add button to cancel update
- remove updated rule and create new rule with updated values

## Related

- [ ] #9799

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html [hunk 1]
```diff
@@ -194,23 +194,61 @@
...
-                    ></td>
+                    <td scope="row" data-ng-if="!rule.errorMessage">
+                        <select
+                            required
+                            id="protocol"
+                            name="protocol"
+                            class="form-control"
+                            data-ng-model="IpGameFirewallCtrl.editRule.protocol"
+                            data-ng-options="IpGameFirewallCtrl.getProtocoleText(protocol) for protocol in IpGameFirewallCtrl.enums.protocols"
+                            data-ng-if="rule.isEditable"
```
Comment (JacquesLarique) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html:205]: this condition could be set on the parent td, this way you wouldn't have to set it on both child input

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html [hunk 2]
```diff
@@ -194,23 +194,61 @@
...
+                            type="number"
+                            data-ng-model="IpGameFirewallCtrl.editRule.ports.from"
+                            class="form-control"
+                            required
+                            id="portFrom"
+                            name="portFrom"
+                            data-ng-min="1"
+                            data-ng-max="65535"
+                            data-ng-pattern="/^[0-9]*$/"
+                            data-ng-if="rule.isEditable"
```
Comment (JacquesLarique) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html:224]: Same as previous comment

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html [hunk 3]
```diff
@@ -194,23 +194,61 @@
...
+                        <input
+                            type="number"
+                            class="form-control"
+                            data-ng-model="IpGameFirewallCtrl.editRule.ports.to"
+                            id="portTo"
+                            name="portTo"
+                            data-ng-min="1"
+                            data-ng-max="65535"
+                            data-ng-pattern="/^[0-9]*$/"
+                            data-ng-if="rule.isEditable"
```
Comment (JacquesLarique) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html:241]: same as previous comment

**Review by JacquesLarique — CHANGES_REQUESTED** [review]

**Review by anooparveti — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html [hunk 4]
```diff
@@ -194,23 +191,62 @@
...
                     ></td>
+                    <td data-ng-if="rule.isEditable && !rule.errorMessage">
+                        <input
+                            type="number"
+                            class="form-control"
+                            data-ng-model="IpGameFirewallCtrl.editRule.ports.to"
+                            id="portTo"
+                            name="portTo"
+                            data-ng-min="1"
+                            data-ng-max="65535"
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html:238]: Can you please maintain min and max in constant and used it here ?

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html [hunk 5]
```diff
@@ -194,23 +191,62 @@
...
+                    <td data-ng-if="rule.isEditable && !rule.errorMessage">
+                        <input
+                            type="number"
+                            data-ng-model="IpGameFirewallCtrl.editRule.ports.from"
+                            class="form-control"
+                            required
+                            id="portFrom"
+                            name="portFrom"
+                            data-ng-min="1"
+                            data-ng-max="65535"
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html:222]: same as here: https://github.com/ovh/manager/pull/9848/files#r1301262617

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html [hunk 6]
```diff
@@ -194,23 +191,62 @@
...
+                            data-ng-pattern="/^[0-9]*$/"
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html:223]: This regular expression can also be mainatined in constant file.

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html [hunk 7]
```diff
@@ -194,23 +191,62 @@
...
+                    <td data-ng-if="rule.isEditable && !rule.errorMessage">
+                        <input
+                            type="number"
+                            class="form-control"
+                            data-ng-model="IpGameFirewallCtrl.editRule.ports.to"
+                            id="portTo"
+                            name="portTo"
+                            data-ng-min="1"
+                            data-ng-max="65535"
+                            data-ng-pattern="/^[0-9]*$/"
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.html:239]: same as here: https://github.com/ovh/manager/pull/9848/files#r1301266509

**Review by sachinrameshn — APPROVED** [review]

**Review by sachinrameshn — CHANGES_REQUESTED** [review]
Can you please change the commit message to include ticket #  and remove other things

Ex: 
feat(universe): title

ref: ticket#
description

Signed-off-by: Stephanie Moallic <stephanie.moallic@corp.ovh.com>

**Review by sachinrameshn — APPROVED** [review]

**Comment by Steffy29** [comment]
This feature is postponed for the moment

**Pull request closed without merging**